include/server_info.h

```diff
@@ -6,14 +6,19 @@
 #include <memory>
 
 #include "config_parser.h"
+#include "request_handler.h"
 
 struct PathInfo {
+
   /* Name must match id in corresponding request handler */
   std::string handler_id;
 
   /* Contains the config for this particular handler */
   NginxConfig* config;
 
+  /* Pointer to the actual handler object */
+  RequestHandler* handler;
+
 };
 
 struct ServerInfo {

```

Comment:
We don't need `ServerInfo` either. It's just a wrapper for the map, so we can just typedef it as `map<string, RequestHandler*>` in `server.h`. It'll map from URI prefix to handler pointer.